src/sentry/nodestore/base.py

```diff
@@ -85,3 +94,46 @@ def generate_id(self):
 
     def cleanup(self, cutoff_timestamp):
         raise NotImplementedError
+
+    def _get_cache_item(self, id):
+        if self.cache and self.sample_rate != 0.0:
+            return self.cache.get(id)
+
+    def _get_cache_items(self, id_list):
+        if self.cache and self.sample_rate != 0.0:
```

Comment:
Any reason for not being consisntent with set_cache_item and calling should_cache(id) ?

Reply:
I forgot we now have a way to check which ones shoudl be cached. Adding.